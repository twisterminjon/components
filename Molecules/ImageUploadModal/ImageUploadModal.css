.imageuploadmodal-dimmer-screen {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: var(--misc__dimmer_bg);

  z-index: var(--image-upload-modal);
}

.imageuploadmodal-wrap {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  background-color: var(--grays_0_1);

  width: 428px;
  padding: 30px;
}

.imageuploadmodal-header {
  display: flex;
  justify-content: space-between;
  align-items: center;

  width: 100%;
  margin-bottom: 30px;
}

.imageuploadmodal-content {
  height: 528px;
}

.imageuploadmodal-text {
  color: var(--base__body_fg);
  font-size: var(--font-base-body--size-xlarge);
  font-weight: var(--font-base--bold);
}

.imageuploadmodal-close {
  height: 42px;
  width: 42px;

  background-color: transparent;
  border-radius: 4px;
}

.imageuploadmodal-close:hover {
  background-color: var(--button_color_hover);
}

.imageuploadmodal-close:focus {
  outline: 1px solid var(--button_color_hover);
}

@media screen and (max-width: 375px) {
  .imageuploadmodal-header {
    padding: 0 30px;
  }
}
