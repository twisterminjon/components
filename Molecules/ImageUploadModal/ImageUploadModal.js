import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { toast } from 'react-toastify'

import IconTimes from '../../Atoms/Icons/IconTimes'
import ImageEditor from '../ImageEditor/ImageEditor'
import FileSelect from '../FileSelect/FileSelect'
import { FILE_NOT_ALLOWED } from '../../../constants'

import './ImageUploadModal.css'

ImageUploadModal.propTypes = {
  /** Can show a loader */
  loading: PropTypes.bool,

  /** Called after modal has closed */
  onClose: PropTypes.func.isRequired,

  /** Called after the save action */
  onSave: PropTypes.func.isRequired,

  /** Can show specific title */
  title: PropTypes.string,
}
ImageUploadModal.defaultProps = {
  loading: false,
  title: 'Provide photo',
}

export default function ImageUploadModal({ title, loading, onSave, onClose }) {
  const [image, setImage] = useState(null)

  const handleSave = imageData => {
    // We need to handle bad images and reset so the picker shows
    if (imageData === FILE_NOT_ALLOWED) {
      toast.warn('Sorry, the file could not be uploaded. Please try another file.')
      setImage(null)
    } else {
      onSave(imageData)
    }
  }

  return (
    <div className="imageuploadmodal-dimmer-screen" data-testid="page-pic-upload">
      <div className="imageuploadmodal-wrap">
        <div className="imageuploadmodal-header">
          <span className="imageuploadmodal-text">{image ? 'Move and scale' : title}</span>
          <button className="button-reset imageuploadmodal-close" onClick={onClose} data-testid="button-close-upload">
            <IconTimes color="var(--base__body_fg)" size={20} />
          </button>
        </div>
        <div className="imageuploadmodal-content">
          {image ? (
            <ImageEditor
              loading={loading}
              file={image}
              onSave={handleSave}
              onClose={onClose}
              showCircularCropping={false}
            />
          ) : (
            <FileSelect onSelected={setImage} onClose={onClose} />
          )}
        </div>
      </div>
    </div>
  )
}
