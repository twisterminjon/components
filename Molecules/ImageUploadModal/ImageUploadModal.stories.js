import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import ImageUploadModal from './ImageUploadModal'

const stories = storiesOf('Provider/v1/Molecules', module)

stories
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .add('ImageUploadModal', () => {
    const title = text('title', 'Some not very long and boring title')
    const loadingVal = boolean('loading', false)

    return (
      <div
        style={{
          backgroundImage: 'url("https://picsum.photos/500/500/?random")',
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'cover',
          position: 'absolute',
          top: 0,
          right: 0,
          bottom: 0,
          left: 0,
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <ImageUploadModal loading={loadingVal} title={title} onClose={action('onClose')} onSave={action('onSave')} />
      </div>
    )
  })
