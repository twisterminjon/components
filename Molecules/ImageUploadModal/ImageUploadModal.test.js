import React from 'react'

import ImageUploadModal from './ImageUploadModal'

describe('ImageUploadModal', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<ImageUploadModal onClose={mockFun} onSave={mockFun} />)
    expect(wrapper).toMatchSnapshot()
  })
})
