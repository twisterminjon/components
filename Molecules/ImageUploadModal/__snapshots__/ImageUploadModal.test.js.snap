// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUploadModal matches the snapshot 1`] = `
<div
  className="imageuploadmodal-dimmer-screen"
  data-testid="page-pic-upload"
>
  <div
    className="imageuploadmodal-wrap"
  >
    <div
      className="imageuploadmodal-header"
    >
      <span
        className="imageuploadmodal-text"
      >
        Provide photo
      </span>
      <button
        className="button-reset imageuploadmodal-close"
        data-testid="button-close-upload"
        onClick={[MockFunction]}
      >
        <IconTimes
          color="var(--base__body_fg)"
          size={20}
        />
      </button>
    </div>
    <div
      className="imageuploadmodal-content"
    >
      <FileSelect
        onClose={[MockFunction]}
        onRejected={[Function]}
        onSelected={[Function]}
      />
    </div>
  </div>
</div>
`;
